src-tauri/src/mod_management/mod.rs

```diff
@@ -53,6 +52,35 @@ pub struct ModJson {
     version: Option<String>,
 }
 
+#[derive(Debug)]
+pub struct TempFile(fs::File, PathBuf);
+
+impl TempFile {
+    pub fn new(file: fs::File, path: PathBuf) -> Self {
+        Self(file, path)
+    }
+
+    pub fn file(&self) -> &fs::File {
+        &self.0
+    }
+}
+
+impl Drop for TempFile {
+    fn drop(&mut self) {
+        println!("path {}", self.1.display());
+
+        _ = dbg!(fs::remove_file(&self.1));
```

Comment:
Maybe use [`log`](https://crates.io/crates/log) crate related functions (`log::info!()` etc) for prints so that we can keep stuff consistent ^^

Reply:
I forgot to remove debug stuff